vumi/transports/httprpc/httprpc.py

```diff
@@ -193,6 +195,16 @@ def setup_transport(self):
             ],
             self.web_port)
 
+        self.status_detect = StatusEdgeDetector()
+
+    def add_status(self, **kw):
+        '''Publishes a status if it is not a repeat of the previously
+        published status.'''
+        status = TransportStatus(**kw)
+        if self.status_detect.check_status(status):
```

Comment:
Given how `publish_status` accepts `**kw`, I wonder if the edge detector shouldn't work the same way? Or at least have the option of being used that way.